lib/analysis/network.py

```diff
@@ -45,26 +44,27 @@ def message_number_graph(log_dict, nicks, nick_same_list, DAY_BY_DAY_ANALYSIS =
                     corrected_nick = util.correctLastCharCR(parsed_nick.group(0)[1:-1])
                     nick_sender = ""
                     nick_receiver = ""
-                    for i in range(config.MAX_EXPECTED_DIFF_NICKS):
-                        if ((i < len(conn_comp_list)) and (corrected_nick in conn_comp_list[i])):  #change nick_same_list to conn_comp_list because conn_comp_list is the main list of all users and nicks now
-                            nick_sender = conn_comp_list[i][0]
-                            break
+                    nick_sender = nick_same_list_to_conn_comp_list(conn_comp_list, corrected_nick)        
```

Comment:
I'm not really sure if this function will work correctly here, since the break might not be executed.
Please confirm this.
Also, check outputs of some other graphs to confirm that you aren't breaking any code.